osu.Game/Online/OsuOnlineStore.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.IO.Stores;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// An <see cref="OnlineStore"/> which proxies external media lookups through osu-web.
+    /// </summary>
+    public class OsuOnlineStore : OnlineStore
+    {
+        private readonly string apiEndpointUrl;
+
+        public OsuOnlineStore(string apiEndpointUrl)
+        {
+            this.apiEndpointUrl = apiEndpointUrl;
+        }
+
+        protected override string GetLookupUrl(string url)
+        {
+            if (Uri.TryCreate(url, UriKind.Absolute, out Uri? uri) && uri.Host.EndsWith(@".ppy.sh", StringComparison.OrdinalIgnoreCase))
+                return url;
+
+            return $@"{apiEndpointUrl}/beatmapsets/discussions/media-url?url={url}";
```

Comment:
not sure if this is a big deal, but it feels wrong to me for lazer to be explicitly depending on a non-api route, even if it will still function properly in this case. and it should probably be updated to not have a uri specific to discussions :thinking:

Reply:
I'm not entirely sure what the concern is with a non-API route. As for the definition of the endpoint looking like it's specific to discussions, that's probably something to bring to @ppy/team-web, if action is deemed necessary and feasible.

Reply:
Yeah I dunno about this.

I believe that endpoint should *only* be used for discussion links (are we somehow limiting this? could an abusive user be using this endpoint to create arbitrary proxied links?). If we're using it for more than this it should be something more generalised.

Can the store be used local to the discussions/comments system? And the default osu!-side online store just blocks all non-ppy resources?

Reply:
Sounds feasible, will update as such.

Reply:
I am so baffled as to how things got as far as this PR without talking about any of the concerns with using the route in question in such an undocumented manner *first*. 

To me this either should have not been ever PR'd, or should have at least been preempted with a "is this even ok to do" sanity check.

Reply:
hm yeah, I'm thinking of changing the url to something more generic. Dunno if api will need its own specific url under `/api` but I suppose it can be added...

Reply:
@bdach I think you've extremely raised the tone up there, I will say I have chose to open this PR so that if it's alright then it's alright and if it's not then the points of why and whatever are discussed appropriately in here with easier access to the diff, period.

Reply:
I'll point out that the media-url proxying on discussions is mainly so someone can't just drop in an image for link tracking or fingerprinting users and only partly for caching if the host falls over. If there's a way to do that without proxying, that would be preferrable.

Reply:
the proxying happens on all user posted images...

media-url thing is specifically because the markdown parsing happens on frontend so it doesn't have the camo key and can't just generate the actual proxied link like in backend.